Login page

khalosmoscato opened the pull request:
## Description

Created Login / Sign Up Page and sub-components needed for it.

- **Related issue:** [Insert issue number or link here]

## Type of Change

Please delete options that are not relevant:

- ✨ New feature, style (non-breaking change that adds functionality)
- ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- 📄 Documentation changes (if none of the above apply)
- 🔧 Refactor, perf, test, build, fix
- 🧹 Chore, CI, Revert

## Checklist

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (if necessary)
- [ ] I have updated the documentation (if necessary)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

## Screenshots (if applicable)

<img width="490" alt="Screenshot 2024-10-21 at 16 53 53" src="https://github.com/user-attachments/assets/348dbf75-958c-45e3-9734-f49ed034f5c5">

khalosmoscato committed:
feat: created authentication page and subcomponents needed for it

khalosmoscato committed:
feat: created new subcomponent for the login and sign up form

khalosmoscato committed:
refactor: moved components in proper folder

william-man commented:
Does the button being left aligned make the login form a bit inconsistent?

william-man commented on src/components/Authentication/BaseForm.tsx:
```diff
@@ -0,0 +1,48 @@
+  }[];
+  buttonLabel: string;
+  onSubmit: (e: React.FormEvent) => void;
+}
```
Can this be moved to the library you've created?

william-man commented on src/components/Authentication/BaseForm.tsx:
```diff
@@ -0,0 +1,48 @@
+      ))}
+
+      <div className="mb-4">
+      <Button label={buttonLabel} onClick={() => onSubmit} buttonType="cardButton" />
```
Does this need an aria label to tell screen readers what it is?

william-man commented on src/components/Authentication/FormInput.tsx:
```diff
@@ -0,0 +1,34 @@
+  id: string;
+  placeholder: string;
+  required?: boolean;
+}
```
Does this need to be moved into the types library?

william-man commented on src/pages/Authentication.tsx:
```diff
@@ -0,0 +1,36 @@
+              {isLogin ? "Don't have an account?" : "Already have an account?"}{" "}
+              <button
+                onClick={toggleForm}
+                className="text-zubiHighlight font-bold underline ml-1"
```
Does this need an aria label?

khalosmoscato committed:
Merge branch 'main' of https://github.com/fac30/PRO04_Frontend_Jack_Khalos_William into login-page

khalosmoscato committed:
refactor: moved button interface to our library and added an aria label props for accessibility

khalosmoscato committed:
chore: button centred for consistency

khalosmoscato committed:
fix: added a sublmit handler function

khalosmoscato committed:
feat: added special background to login in the navbar menu and moved interfaces to our library

khalosmoscato commented:
![image](https://github.com/user-attachments/assets/cc4743ca-8966-490e-87ab-12f7db6fa577)
@william-man centred the button and also added aria label for accessibility. 
Moved interface to our library.
@jackcasstlesjones made the `Log In` link stand out in the menu by adding a yellow button-like background, which means I also had to change the hoover effect otherwise the text would become yellow on yellow, but not it becomes white.

khalosmoscato committed:
chore: moved interface to library

khalosmoscato commented:
@jackcasstlesjones @william-man can one of you help me with best practice of sanitise and validate before we merge this to main. Should we use an external libraries, or implement our own validation functions for each input field?

khalosmoscato committed:
docs: updated readme

khalosmoscato committed:
feat: login frontend and backend work

khalosmoscato committed:
feat: once logged in, redirects to homepage

khalosmoscato committed:
feat: when user logs in, it updated navbar links (+ option lo log out)

khalosmoscato committed:
docs: updated readme

khalosmoscato commented:
Error message when incorrect login credentials are entered:
<img width="471" alt="Screenshot 2024-10-23 at 14 43 10" src="https://github.com/user-attachments/assets/ba15257d-b7a3-4c14-abf5-6346d42bcd28">

khalosmoscato commented:
When login is successful, redirects to home page and `My Account` becomes `Student Dashboard` or `Tutor Dashboard`.

This is a bit more future proof, because when we will have the tutors login in the future, they will have access to a different dashboard then students. 

I created an interface that defines the user role:

`export interface User {
  name?: string; // Optional in case no name is provided
  email: string;
  role: 'student' | 'tutor'; 
}`

<img width="544" alt="Screenshot 2024-10-23 at 14 43 37" src="https://github.com/user-attachments/assets/bd5e20ce-42a0-4d6d-87ee-a32828ff9bce">

khalosmoscato commented:
I need to: improve the logout functionality to not redirect the user back to the login page, but refresh the homepage instead.

william-man reviewed (approved):
lgtm
